web/src/TopBar.tsx

```diff
@@ -3,24 +3,27 @@ import { ResourceView } from "./types"
 import "./TopBar.scss"
 import SailInfo from "./SailInfo"
 import TabNav from "./TabNav"
+import PathBuilder from "./PathBuilder"
 
 type TopBarProps = {
   previewUrl: string
   logUrl: string
   errorsUrl: string
   resourceView: ResourceView
+  pathBuilder: PathBuilder
```

Comment:
it might be easier to pass in a previewUrl after it's gone thru PathBuilder? this is fine too tho

Reply:
damnit you're totally right